### ISSUE TITLE
💚 Validation action stops with error "Register-AzModule : The type initializer for 'Microsoft.Azure.Commands.Common.AzModule' threw an exception."

### ISSUE THREAD

**Issue by FH-Inway** [d365collaborative/d365fo.tools#784]
Sometime in the last three weeks, the action to validate pull requests stopped working. Without any changes since the last successful run, it now stops with the error message: 

> Register-AzModule : The type initializer for 'Microsoft.Azure.Commands.Common.AzModule' threw an exception.

This is the last successful run of the action on November 10th 2023: https://github.com/d365collaborative/d365fo.tools/actions/runs/6823750193

The first failing run was on November 29th 2023:
https://github.com/d365collaborative/d365fo.tools/actions/runs/7022751791
This was for a pull request, but the error was not caused by the changes in the pull request. See the run https://github.com/FH-Inway/d365fo.tools/actions/runs/7050406675 in my fork that was done on the unchanged development branch, but also fails with the same error.

The legacy Azure DevOps pipelines that do the same validation also fail.

The error happens here:
https://github.com/d365collaborative/d365fo.tools/blob/e197350b29923c053b251521391801766ff29207/d365fo.tools/tests/pester-PSScriptAnalyzer.ps1#L16-L20

Full error log:

> 2023-11-29T05:56:56.1736174Z ##[group]Run build\vsts-validate-psscriptanalyzer.ps1 -TestPublic $true -TestInternal $false
2023-11-29T05:56:56.1737132Z [36;1mbuild\vsts-validate-psscriptanalyzer.ps1 -TestPublic $true -TestInternal $false[0m
2023-11-29T05:56:56.1789185Z shell: C:\Windows\System32\WindowsPowerShell\v1.0\powershell.EXE -command ". '{0}'"
2023-11-29T05:56:56.1789751Z ##[endgroup]
2023-11-29T05:56:56.7324654Z Working on the machine named: fv-az1564-31
2023-11-29T05:56:56.7332951Z The user running is: runneradmin
2023-11-29T05:56:56.7399825Z Importing PSFramework
2023-11-29T05:56:58.7029971Z Importing PSScriptAnalyzer
2023-11-29T05:57:00.9049787Z Importing Azure.Storage
2023-11-29T05:57:02.9594306Z Importing AzureAd
2023-11-29T05:57:03.0421585Z Importing PSNotification
2023-11-29T05:57:07.3774527Z Importing PSOAuthHelper
2023-11-29T05:57:07.6425521Z Importing ImportExcel
2023-11-29T05:57:09.1931889Z [05:57:09][pester-PSScriptAnalyzer.ps1] Starting Tests
2023-11-29T05:57:09.2012850Z [05:57:09][pester-PSScriptAnalyzer.ps1] Importing Module
2023-11-29T05:57:10.6797235Z WARNING: Both Az and AzureRM modules were detected on this machine. Az and AzureRM modules cannot be imported in the 
2023-11-29T05:57:10.6798952Z same session or used in the same script or runbook. If you are running PowerShell in an environment you control you can
2023-11-29T05:57:10.6800907Z  use the 'Uninstall-AzureRm' cmdlet to remove all AzureRm modules from your machine. If you are running in Azure 
2023-11-29T05:57:10.6802540Z Automation, take care that none of your runbooks import both Az and AzureRM modules. More information can be found 
2023-11-29T05:57:10.6803853Z here: https://aka.ms/azps-migration-guide
2023-11-29T05:57:12.5823874Z Register-AzModule : The type initializer for 'Microsoft.Azure.Commands.Common.AzModule' threw an exception.
2023-11-29T05:57:12.5825314Z At C:\Users\runneradmin\Documents\WindowsPowerShell\Modules\Az.Storage\6.0.0\Storage.Autorest\Az.Storage.psm1:49 
2023-11-29T05:57:12.5826292Z char:13
2023-11-29T05:57:12.5826622Z +   $VTable = Register-AzModule
2023-11-29T05:57:12.5826929Z +             ~~~~~~~~~~~~~~~~~
2023-11-29T05:57:12.5827490Z     + CategoryInfo          : CloseError: (:) [Register-AzModule], TypeInitializationException
2023-11-29T05:57:12.5828413Z     + FullyQualifiedErrorId : Microsoft.Azure.Commands.Common.RegisterAzModule
2023-11-29T05:57:12.5828899Z  
2023-11-29T05:57:17.2585125Z ##[error]Process completed with exit code 1.
>
> Error: Process completed with exit code 1.

**Comment by Splaxi** [comment]
The agents running will most likely have a new version of a specific module, which is autoloaded. When we load our stuff, it fails. I'll see if I can find time to dig into the details of this...

**Comment by FH-Inway** [comment]
Think I found the culprit, see #785. Turns out the PSScriptAnalyzer tests were still running with the Azure.Storage module instead of Az.Storage. Not sure though why this only now results in an error. While the image of the agent runner has changed, I don't see any PowerShell related changes that could have caused this.

https://github.com/actions/runner-images/commit/ca9f1bc9450b8a5523f9c53a5b49d4db3ddd6acf#diff-d3119a278ce9ce80fc4b5af63a1112fb3a044aec3e6002d838e8eb26d7a3a499R562-R581
